### ISSUE TITLE
Trace throws exception when depCache is defined in config.js

### ISSUE THREAD

**Issue by brillout** [jspm/jspm-cli#1044]
Following code
```
var jspm = require('jspm');
new jspm.Builder().trace('js/main');
```
throws an exception because;
  - following dependency is defined in the depCache;
```
// js/config.js
System.config({
  depCache: {
    "js/main.js": [
      "js/hello.js"
    ]
  }});
```
  - and the source code of `js/hello.js` contains a reference to `document`;
```
// js/hello.js
document.body.innerHTML += '<div>Hello from ES2015</div>';
```

The problem seems to be that the depCache makes SystemJS pre-load and pre-evaluate `js/hello.js` upfront but node doesn't have a DOM and throws a `ReferenceError` exception upon `new Function('document')()`.

The exception with stack trace;
```
Unhandled rejection Error: ReferenceError: file:///home/romu/code/demo/src/js/hello.js:1
document.body.innerHTML += '<div>Hello from ES2015</div>';
^
document is not defined
        Evaluating file:///home/romu/code/demo/src/js/hello.js
        Error loading file:///home/romu/code/demo/src/js/hello.js
    at createImportPromise (/home/romu/code/demo/node_modules/jspm/node_modules/systemjs-builder/node_modules/systemjs/dist/system.src.js:735:43)
    at /home/romu/code/demo/node_modules/jspm/node_modules/systemjs-builder/node_modules/systemjs/dist/system.src.js:798:47
From previous event:
    at SystemJSNodeLoader.Loader.import (/home/romu/code/demo/node_modules/jspm/node_modules/systemjs-builder/node_modules/systemjs/dist/system.src.js:790:8)
    at SystemJSNodeLoader.import (/home/romu/code/demo/node_modules/jspm/node_modules/systemjs-builder/node_modules/systemjs/dist/system.src.js:1418:25)
    at SystemJSNodeLoader.locate (/home/romu/code/demo/node_modules/jspm/node_modules/systemjs-builder/node_modules/systemjs/dist/system.src.js:3874:27)
    at /home/romu/code/demo/node_modules/jspm/node_modules/systemjs-builder/lib/trace.js:42:57
From previous event:
    at Trace.getLoadRecord (/home/romu/code/demo/node_modules/jspm/node_modules/systemjs-builder/lib/trace.js:21:4)
    at /home/romu/code/demo/node_modules/jspm/node_modules/systemjs-builder/lib/trace.js:100:17
    at processImmediate [as _immediateCallback] (timers.js:371:17)
From previous event:
    at Trace.getAllLoadRecords (/home/romu/code/demo/node_modules/jspm/node_modules/systemjs-builder/lib/trace.js:94:4)
    at /home/romu/code/demo/node_modules/jspm/node_modules/systemjs-builder/lib/builder.js:273:24
    at lib$rsvp$$internal$$tryCatch (/home/romu/code/demo/node_modules/jspm/node_modules/rsvp/dist/rsvp.js:493:16)
    at lib$rsvp$$internal$$invokeCallback (/home/romu/code/demo/node_modules/jspm/node_modules/rsvp/dist/rsvp.js:505:17)
    at lib$rsvp$$internal$$publish (/home/romu/code/demo/node_modules/jspm/node_modules/rsvp/dist/rsvp.js:476:11)
    at lib$rsvp$asap$$flush (/home/romu/code/demo/node_modules/jspm/node_modules/rsvp/dist/rsvp.js:1198:9)
    at doNTCallback0 (node.js:408:9)
    at process._tickCallback (node.js:337:13)
```

**Comment by guybedford** [comment]
Thanks, I've added a fix in https://github.com/jspm/jspm-cli/commit/6e35425afe78fb0b61c8314e5aa3329852122839.

**Comment by guybedford** [comment]
Released in 0.16.1.
